lib/special_route_publisher.rb

```diff
@@ -1,38 +1,83 @@
 require "gds_api/publishing_api/special_route_publisher"
 
 class SpecialRoutePublisher
-  def initialize(publisher_options)
-    @publishing_api = publisher_options[:publishing_api]
-    @publisher = GdsApi::PublishingApi::SpecialRoutePublisher.new(publisher_options)
+  def initialize
+    @publishing_api = Services.publishing_api
+    logger = Rails.logger
+    @publisher = GdsApi::PublishingApi::SpecialRoutePublisher.new(logger: logger, publishing_api: @publishing_api)
   end
 
-  def publish(route_type, route)
-    @publisher.publish(
-      route.merge(
-        publishing_app: "collections-publisher",
-        rendering_app: "collections",
-        type: route_type,
-        public_updated_at: Time.zone.now.iso8601,
-        update_type: "major",
-      ),
-    )
+  def publish(route)
+    default_options = {
+      publishing_app: "collections-publisher",
+      rendering_app: "collections",
+      type: "exact",
+      update_type: "major",
+    }
+
+    @publisher.publish(default_options.merge(route))
   end
 
   def unpublish(content_id, options)
     @publishing_api.unpublish(content_id, options)
   end
 
+  def self.find_route(base_path)
+    routes.find { |route| route[:base_path] == base_path }
+  end
+
   def self.routes
-    {
-      exact: [
-        {
-          document_type: "answer",
-          content_id: "bb986a97-3b8c-4b1a-89bf-2a9f46be9747",
-          base_path: "/eubusiness",
-          title: "Trade with the UK from 1 January 2021 as a business based in the EU",
-          description: "The UK has left the EU. On 31 December 2020 the UK will leave the EU single market and customs union. From 1 January 2021 the rules for trading with the UK will change.",
-        },
-      ],
-    }
+    [
+      {
+        document_type: "answer",
+        content_id: "bb986a97-3b8c-4b1a-89bf-2a9f46be9747",
+        base_path: "/eubusiness",
+        locale: "en",
+        title: "Trade with the UK from 1 January 2021 as a business based in the EU",
+        description: "The UK has left the EU. On 31 December 2020 the UK will leave the EU single market and customs union. From 1 January 2021 the rules for trading with the UK will change.",
+      },
+      {
+        content_id: "bb986a97-3b8c-4b1a-89bf-2a9f46be9747",
+        base_path: "/eubusiness.de",
+        locale: "de",
+        title: "Trade with the UK from 1 January 2021 as a business based in the EU",
+        description: "The UK has left the EU. On 31 December 2020 the UK will leave the EU single market and customs union. From 1 January 2021 the rules for trading with the UK will change.",
+      },
+      {
+        content_id: "bb986a97-3b8c-4b1a-89bf-2a9f46be9747",
+        base_path: "/eubusiness.es",
+        locale: "es",
+        title: "Trade with the UK from 1 January 2021 as a business based in the EU",
```

Comment:
Should these be the translated titles and descriptions? I can push the translations to this branch if so.

Reply:
They can be, and it's preferable that they are so that we're consistent with use of the locale. I don't think they'll be used anywhere yet though, so it doesn't need to block us. Thanks!